owasp-top10-2016-mobile/m2/cool_games/README.md

```diff
@@ -0,0 +1,178 @@
+# Cool Games
+
+<p align="center">
+    <img src="images/log_in.png"/>
+</p>
+
+Cool Games is a Flutter powered mobile app that serves as the mobile version of [GamesIRADOS.com](https://github.com/globocom/secDevLabs/tree/master/owasp-top10-2017-apps/a10/games-irados). It is built with a mobile front-end a Golang powered back-end.
+
+## Index
+
+- [Definition](#what-is-insecure-authentication)
+- [Setup](#setup)
+- [Attack narrative](#attack-narrative)
+- [Objectives](#secure-this-app)
+- [Solutions](#pr-solutions)
+- [Contributing](#contributing)
+
+## What is Insecure Data Storage?
+
+Insecure data storage vulnerabilities occur when development teams assume that users or malware will not have access to a mobile device’s filesystem and subsequent sensitive information in data-stores on the device. Filesystems are easily accessible. Organizations should expect a malicious user or malware to inspect sensitive data stores.
+
+The main goal of this app is to discuss how **Insecure Data Storage** vulnerabilities can be exploited and to encourage developers to send secDevLabs Pull Requests on how they would mitigate these flaws.
+
+## Setup
+
+Before we start, it's important to mention that this app's code is divided into two parts: a back-end server, which can be found in the `server` folder, and the mobile app's code itself in the `mobile` folder. In order for the app to run as intended, the back-end server must be up and running, but no need to worry, we'll walk you through it! 😁
+
+To start this intentionally **insecure application**, you will need [Flutter][VS-Code-Flutter], [Docker][Docker Install] and [Docker Compose][Docker Compose Install]. After forking [secDevLabs](https://github.com/globocom/secDevLabs), you'll need to start the server, which can be done through the commands:
+
+### Start server commands:
+
+```sh
+cd secDevLabs/owasp-top10-2016-mobile/m2/cool_games/server
+```
+
+```sh
+make install
+```
+
+### Start app commands:
+
+**Note**: It is important to mention, that you should have an emulator up and running or an available device to run the app in. For instructions on how to set up an emulator, click [here][VS-Code-Flutter].
+
+After properly setting up your editor and emulator/device to run Flutter apps, open the `main.dart` file with VS Code. You can find it's path here:
+
+```
+secDevLabs/owasp-top10-2016-mobile/m2/cool_games/mobile/lib/main.dart
+```
+
+Now, all you need to do is to click `Run > Run Without Debugging` on VS Code top menu bar and it should begin building the app to launch it in no time!
+
+Then, you should see Cool Games' app launch successfully in the emulator/device you're using! 📲
+
+**Note**: In case an Android licenses errors shows up, you may need to run the following command on your terminal and accept them:
+
+```sh
+flutter doctor --android-licenses
+```
+
+## Get to know the app 📝
+
+To properly understand how this application works, you can follow these simple steps:
+
+- Register a new user. 👩‍💻
```

Comment:
I tried to register my user with username "eu" and password "12345" but couldn't log in with this credential and didn't receive any error during the registration. After I tried with a longer username and another password it worked.

Reply:
I'm sorry, @joserenatosilva, but I couldn't reproduce this issue. I created an user with `eu` as username and `12345` as password and logged in without any problems.

Reply:
Maybe it was something wrong with my setup. When I tried it again it worked. Thanks for double-checking 🙃.